### PR TITLE
Add Duplicate Check for Question Service

### PR DESCRIPTION
**Pull request by Jnjy**

### CODE REVIEW

File: backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java [hunk 1]
```diff
@@ -84,6 +84,13 @@ public List<QuestionDto> getQuestionsByCategory(String category) {
     public QuestionDto createQuestion(QuestionDto questionDto) {
         validateQuestion(questionDto);
 
+        // Check if the question already exists in the database
+        List<Question> existingQuestions = questionRepository.findByTitle(questionDto.getTitle());
+
+        if (!existingQuestions.isEmpty()) {
+            throw new InvalidQuestionException("Duplicate question found with title: " + questionDto.getTitle());
+        }
+
```
Comment (zsh-eng) [inline backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java]: For the DTO, should we strip leading and trailing whitespace before inserting to the database? 

Also should we do a case insensitive match?

Comment (Jnjy) [reply backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java:93]: Yeap, good suggestions, let me add it in!

Comment (Jnjy) [reply backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java]: @zsh-eng Thanks for the suggestion, I will allow for case insensitive match! However, will it be possible the leading and trailing whitespace to be handled by frontend?

Comment (Jnjy) [reply backend/service-question/src/main/java/g55/cs3219/backend/questionservice/service/QuestionService.java]: > @zsh-eng Thanks for the suggestion, I will allow for case insensitive match! However, will it be possible the leading and trailing whitespace to be handled by frontend?

I have added both to backend. I will change it to use an Enum class instead, will make a separate PR.

**Pull request merged**